components/eam/src/physics/cam/modal_aer_opt.F90

```diff
@@ -700,439 +620,254 @@ subroutine modal_aero_sw(list_idx, dt, state, pbuf, nnite, idxnite, is_cmip6_vol
-                     abslip(i)  = (abslip(i) + absh2o*hygrolip(i)/sumhygro)/sumabs
-#endif
-
-                     aodabsbc(i)    = aodabsbc(i) + absbc(i)*dopaer(i)*(1.0_r8-palb(i))
-
-                     aodc           = (absdust(i)*(1.0_r8 - palb(i)) + palb(i)*scatdust(i))*dopaer(i)
-                     dustaod(i)     = dustaod(i) + aodc
-
-                     aodc           = (absso4(i)*(1.0_r8 - palb(i)) + palb(i)*scatso4(i))*dopaer(i)
-                     so4aod(i)      = so4aod(i) + aodc
-
-                     aodc           = (abspom(i)*(1.0_r8 - palb(i)) + palb(i)*scatpom(i))*dopaer(i)
-                     pomaod(i)      = pomaod(i) + aodc
-
-                     aodc           = (abssoa(i)*(1.0_r8 - palb(i)) + palb(i)*scatsoa(i))*dopaer(i)
-                     soaaod(i)      = soaaod(i) + aodc
-
-                     aodc           = (absbc(i)*(1.0_r8 - palb(i)) + palb(i)*scatbc(i))*dopaer(i)
-                     bcaod(i)       = bcaod(i) + aodc
-
-                     aodc           = (absseasalt(i)*(1.0_r8 - palb(i)) + palb(i)*scatseasalt(i))*dopaer(i)
-                     seasaltaod(i)  = seasaltaod(i) + aodc
-
-#if ( defined MODAL_AERO_4MODE_MOM )
-                     aodc           = (absmom(i)*(1.0_r8 - palb(i)) + palb(i)*scatmom(i))*dopaer(i)
-                     momaod(i)  = momaod(i) + aodc
-
-#elif ( defined MODAL_AERO_9MODE )
-                     aodc           = (abspoly(i)*(1.0_r8 - palb(i)) + palb(i)*scatpoly(i))*dopaer(i)
-                     polyaod(i)  = polyaod(i) + aodc
-
-                     aodc           = (absprot(i)*(1.0_r8 - palb(i)) + palb(i)*scatprot(i))*dopaer(i)
-                     protaod(i)  = protaod(i) + aodc
-
-                     aodc           = (abslip(i)*(1.0_r8 - palb(i)) + palb(i)*scatlip(i))*dopaer(i)
-                     lipaod(i)  = lipaod(i) + aodc
-#endif
+                     scath2o  = watervol(icol)*real(crefwsw(isw))
+                     absh2o   = -watervol(icol)*aimag(crefwsw(isw))
+                     sumscat  = scatso4(icol) + scatpom(icol) + scatsoa(icol) + scatbc(icol) + &
+                                scatdust(icol) + scatseasalt(icol) + scath2o + &
+                                scatmom(icol)
+                     sumabs   = absso4(icol) + abspom(icol) + abssoa(icol) + absbc(icol) + &
+                                absdust(icol) + absseasalt(icol) + absh2o + &
+                                absmom(icol)
+                     sumhygro = hygroso4(icol) + hygropom(icol) + hygrosoa(icol) + hygrobc(icol) + &
+                                hygrodust(icol) + hygroseasalt(icol) + &
+                                hygromom(icol)
+
+                     call update_aod_spec ( icol, scath2o, absh2o, & ! in
+                           sumhygro, sumscat, sumabs, & ! in
+                           hygrodust(icol), palb(icol), dopaer(icol), & ! in
+                           scatdust, absdust, dustaod ) ! inout
```

Comment:
Just like you are sending `hygrodust(icol), palb(icol), dopaer(icol),`, if you send `scatdust(icol), absdust(icol), dustaod(icol)`, we won't need to send `icol`. That way the subroutine will look like it is operating on only one column.